frontend/appflowy_flutter/lib/plugins/database/grid/application/filter/select_option_filter_bloc.dart

```diff
@@ -1,146 +0,0 @@
-import 'package:appflowy/plugins/database/grid/presentation/widgets/filter/filter_info.dart';
-import 'package:appflowy_backend/protobuf/flowy-database2/select_option_filter.pbserver.dart';
-import 'package:appflowy_backend/protobuf/flowy-database2/util.pb.dart';
-import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:freezed_annotation/freezed_annotation.dart';
-
-part 'select_option_filter_bloc.freezed.dart';
-
-class SelectOptionFilterEditorBloc
-    extends Bloc<SelectOptionFilterEditorEvent, SelectOptionFilterEditorState> {
-  SelectOptionFilterEditorBloc({
-    required this.filterInfo,
-    required this.delegate,
-  })  : _filterBackendSvc = FilterBackendService(viewId: filterInfo.viewId),
-        _listener = FilterListener(
-          viewId: filterInfo.viewId,
-          filterId: filterInfo.filter.id,
-        ),
-        super(SelectOptionFilterEditorState.initial(filterInfo)) {
-    _dispatch();
-  }
-
-  final FilterInfo filterInfo;
-  final FilterBackendService _filterBackendSvc;
-  final FilterListener _listener;
-  final SelectOptionFilterDelegate delegate;
-
-  void _dispatch() {
-    on<SelectOptionFilterEditorEvent>(
-      (event, emit) async {
-        event.when(
-          initial: () {
-            _startListening();
-            _loadOptions();
-          },
-          updateCondition: (SelectOptionFilterConditionPB condition) {
-            _filterBackendSvc.insertSelectOptionFilter(
-              filterId: filterInfo.filter.id,
-              fieldId: filterInfo.fieldInfo.id,
-              condition: condition,
-              optionIds: state.filter.optionIds,
-              fieldType: state.filterInfo.fieldInfo.fieldType,
-            );
-          },
-          updateContent: (List<String> optionIds) {
-            _filterBackendSvc.insertSelectOptionFilter(
-              filterId: filterInfo.filter.id,
-              fieldId: filterInfo.fieldInfo.id,
-              condition: state.filter.condition,
-              optionIds: optionIds,
-              fieldType: state.filterInfo.fieldInfo.fieldType,
-            );
-          },
-          delete: () {
-            _filterBackendSvc.deleteFilter(
-              fieldId: filterInfo.fieldInfo.id,
-              filterId: filterInfo.filter.id,
-            );
-          },
-          didReceiveFilter: (FilterPB filter) {
-            final filterInfo = state.filterInfo.copyWith(filter: filter);
-            final selectOptionFilter = filterInfo.selectOptionFilter()!;
-            emit(
-              state.copyWith(
-                filterInfo: filterInfo,
-                filter: selectOptionFilter,
-              ),
-            );
-          },
-          updateFilterDescription: (String desc) {
-            emit(state.copyWith(filterDesc: desc));
-          },
-        );
-      },
-    );
-  }
-
-  void _startListening() {
-    _listener.start(
-      onUpdated: (filter) {
```

Comment:
without this listener, how to refresh the UI

Reply:
Filter editor bloc listens to entire set of filter changes. For any database view, I think the number of filters will never be too high and I use BlocSelectors to reduce widget rebuilds